drivers/wifi/esp_at/esp.c

```diff
@@ -319,7 +319,7 @@ MODEM_CMD_DEFINE(on_cmd_cwjap)
 	}
 
 	strncpy(status->ssid, ssid, sizeof(status->ssid));
-	status->ssid_len = strlen(status->ssid);
+	status->ssid_len = MIN(strlen(status->ssid), sizeof(status->ssid));
```

Comment:
Maybe just use `strnlen()`?

Reply:
Sorry, I haven't had time to follow all the latest libc/posix discussions, I believe there are still a few open questions regarding the way how to define_POSIX_C_SOURCE aren't there?
I have now added the definition in the source file, to be able to build on the native_sim target, but that way the compliance check fails.
Should I revert to the initial version or has there already been reached an agreement on how this should be handled?

Reply:
No agreement yet, but I would consider this compliance error a false positive.